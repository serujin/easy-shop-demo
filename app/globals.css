@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
