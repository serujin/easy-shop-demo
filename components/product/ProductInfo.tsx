import React from "react";
import Text from "../basic/Text/Text";
import Ratings from "../ratings/Ratings";
import Button from "../basic/Button";

type ProductInfoProps = {
  description: string;
  name: string;
  rating: number;
  reviews: number;
};

const buttonOnClickHandler = () => console.log("Ordering samples...");

export default function ProductInfo(props: ProductInfoProps) {
  const { description, name, rating, reviews } = props;
  return (
    <div>
      <Ratings title={name} rating={rating} reviews={reviews} />
      <Text text={description} />
      <Button text={"Order samples"} onClick={buttonOnClickHandler} />
    </div>
  );
}
