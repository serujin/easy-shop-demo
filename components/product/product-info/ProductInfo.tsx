import React from "react";
import Ratings from "../../ratings/Ratings";
import styles from "./ProductInfo.module.css";

type ProductInfoProps = {
  description: string;
  name: string;
  rating: number;
  reviews: number;
};

const buttonOnClickHandler = () => console.log("Ordering samples...");

export default function ProductInfo(props: ProductInfoProps) {
  const { description, name, rating, reviews } = props;
  return (
    <div className={styles.productInfoContainer}>
      <Ratings title={name} rating={rating} reviews={reviews} />
      <p className={styles.description}>{description}</p>
      <button
        className={styles.orderSamplesButton}
        onClick={buttonOnClickHandler}
      >
        Order samples
      </button>
    </div>
  );
}
