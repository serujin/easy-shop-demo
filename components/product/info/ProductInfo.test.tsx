import React from 'react'
import { render } from '@testing-library/react'
import ProductInfo from './ProductInfo'
import { describe, expect, it } from '@jest/globals'

describe('ProductInfo component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ProductInfo
        description="description"
        name="name"
        rating={4.5}
        reviews={1000}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
