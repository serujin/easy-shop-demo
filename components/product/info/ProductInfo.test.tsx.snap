// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo component with samples should render correctly 1`] = `
<div>
  <div
    class="productInfoContainer"
  >
    <div
      class="container"
    >
      <div
        class="titleContainer"
      >
        <p
          class="normal"
        >
          name
        </p>
      </div>
      <div
        class="reviewsContainer normal"
      >
        <div
          class="starsContainer"
        >
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Half star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/half-star.svg"
            style="color: transparent;"
            width="17"
          />
        </div>
        <p>
          1,000 reviews
        </p>
      </div>
    </div>
    <p
      class="description"
    >
      description
    </p>
    <button
      class="orderSamplesButton"
    >
      Order samples
    </button>
  </div>
</div>
`;

exports[`ProductInfo component without samples should render correctly 1`] = `
<div>
  <div
    class="productInfoContainer"
  >
    <div
      class="container"
    >
      <div
        class="titleContainer"
      >
        <p
          class="normal"
        >
          name
        </p>
      </div>
      <div
        class="reviewsContainer normal"
      >
        <div
          class="starsContainer"
        >
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Full star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/full-star.svg"
            style="color: transparent;"
            width="17"
          />
          <img
            alt="Half star"
            class="starItem"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/ratings/half-star.svg"
            style="color: transparent;"
            width="17"
          />
        </div>
        <p>
          1,000 reviews
        </p>
      </div>
    </div>
    <p
      class="description"
    >
      description
    </p>
  </div>
</div>
`;
